src/common/ceph_context.h

```diff
@@ -261,6 +262,15 @@ class CephContext {
   void notify_pre_fork();
   void notify_post_fork();
 
+  void set_mon_addrs(const std::vector<entity_addrvec_t>& in) {
+    auto ptr = std::make_shared<std::vector<entity_addrvec_t>>(in);
+    atomic_store_explicit(&_mon_addrs, std::move(ptr), std::memory_order_relaxed);
+  }
+  std::shared_ptr<std::vector<entity_addrvec_t>> get_mon_addrs() const {
+    auto ptr = atomic_load_explicit(&_mon_addrs, std::memory_order_relaxed);
```

Comment:
@batrick isn't there a potential race where post the atomic fetch of the pointer and prior to incrementing the ref count on the shared pointer, a set_mon_addrs updates the existing pointer, and hence ends up destroying the fetched pointer (as at that time there was only one ref)?

Reply:
No, the atomic_load_explicit increments the refcount on the control block for the shared_ptr before returning it. So there's no chance of a race.

Reply:
"That's one smart pointer!" :wink: Thanks